Math/NumberTheory/EisensteinIntegers.hs

```diff
@@ -44,19 +56,16 @@ instance Show EisensteinInteger where
 instance Num EisensteinInteger where
     (+) (a :+ b) (c :+ d) = (a + c) :+ (b + d)
     (*) (a :+ b) (c :+ d) = (a * c - b * d) :+ (b * c + a * d - b * d)
-    -- An Eisenstein integer @a :+ b@, with @a, b@ integers, can we written as
-    -- @(2*a - b) / 2 + ((b * sqrt 3) * ι) / 2@, but this number is in the
-    -- same quadrant as @(2*a - b) / 2 + (b * ι) / 2@, and this one in the
-    -- same as @(2*a - b) + b * ι@. Divisions or floating points are not
-    -- necessary.
-    abs z@(x :+ y) = abs' (2*x - y) x
+    abs z@(a :+ b)
+        | a == 0 && b == 0                                  = z            -- origin
```

Comment:
@Bodigrim There is probably a better way to do this. I did not want to use `phase` here because it brings `Double`'s precision issues as the numbers get larger.

`abs $ 10^n :+ (10^n - 1)` will probably give the wrong result for some large `n`, but for now it works. I'll think of a way to do this check without phases later.

Reply:
@Bodigrim I derived those bounds as well, but I got wrong results the first time because I was multiplying by the wrong unit (i.e. perfoming an incorrect rotation in the plane), so I decided to make it work first with `Data.Complex.phase` and then think about it some more.